zulipterminal/ui_tools/views.py

```diff
@@ -1663,6 +1667,7 @@ def keypress(self, size: urwid_Size, key: str) -> str:
             self.allow_update_emoji_list = True
         if is_command_key("SEARCH_EMOJIS", key) and self.allow_update_emoji_list:
             self.allow_update_emoji_list = False
+            self.is_popup_search_open = True
```

Comment:
No, this is not because of the above approach. This is actually a fix for a bug that you reported(some keys i.e. j,k,G not getting typed in the search-box). The bug was present because the `PopupView` intercepted some direction keys(both primary and secondary) and converted them to primary(i.e. "up", "down" etc.), thus preventing the secondary letter keys from getting typed in the search box. Hence, I added the above parameter in `__init__()`. See commit f05c892 for the bugfix.

The reason I can't use something like `is_in_editor_mode` is because it is an attribute of `PanelSearchBox` which can't be accessed in `PopupView`(since its the parent of `EmojiPickerView`). You can see that we do check `is_in_editor_mode` in `ENTER` keypress events in `EmojiPickerView`, but we cant prevent the `PopupView` from capturing secondary letter key. This seems to be the only approach feasible to me. :)

Reply:
Oh ok.
But this kills the abstraction of the class right? PopupView now knows that there is a child that uses a searchbox. PopupView shoulld know nothing about its children. Maybe you could change the name to `disable_keys`? Or maybe you can simply override it from the keypress of EmojiView? selectively call super()?
Also if you have access to controller, you get access to `is_in_editor_mode` right?

Reply:
Right, I do have access to `controller`, so we could use that. I've added a comment down below to help me decide what approach should I follow here.